index.ts

```diff
@@ -1,8 +1,9 @@
 import process from "process"
 import ts from "typescript"
 import tsNode from "ts-node"
-
-import { helloWorld } from "./Util"
+// Typescript does not like rewriting Specifiers so you 
+// target.js as that will be the identifier of the run able result
+import { helloWorld } from "./Util.js" 
```

Comment:
I am not sure I understand. Not having an extension work as expected using `--experimental-specifier-resolution=node` for NodeJS 16 and NodeJS 18. 

You can confirm this with this automated tests I just added to the repo: 
https://github.com/felipeplets/esm-examples/actions/runs/3584330368/jobs/6030862023

Reply:
ah you are correct that flag corrects that but again if you produce modules that way they will not work on the web without import maps to support bare specifiers on the web.

Reply:
This repo does not have a working example for web projects to run using ESM at this point in time. If that is your requirement feel free to open an issue requesting it.